composer.json

```diff
@@ -26,15 +28,29 @@
     },
     "sort-packages": true
   },
+  "extra": {
+    "downloads": {
+      "phpstan": {
+        "url": "https://github.com/phpstan/phpstan/releases/latest/download/phpstan.phar",
+        "path": "vendor/bin/phpstan",
+        "type": "phar"
+      }
+    }
+  },
   "autoload": {
     "psr-4": {
       "Google\\Web_Stories\\": "includes"
     }
   },
   "autoload-dev": {
     "psr-4": {
-      "Google\\Web_Stories\\Tests\\": "tests/phpunit/includes"
+      "Google\\Web_Stories\\Tests\\": "tests/phpunit/includes",
+      "PHPStan\\WordPress\\": "tests/phpstan/src"
     }
   },
-  "prefer-stable": true
+  "prefer-stable": true,
+  "scripts": {
```

Comment:
This change requires composer to installed locally for this to run. Do we need to update docs?

Reply:
Update docs where?

Not sure I follow...

PHPStan is a composer package, and to install composer packages you need composer. To build the plugin locally you need composer too.

No change in requirements there.

Reply:
But PHPStan only runs on phpstan 7.2? Why not just install it conditionally? https://github.com/composer/composer/issues/4168#issuecomment-113547922

Reply:
We recommend contributors to run PHP 7.2 or higher, see https://github.com/google/web-stories-wp/blob/main/docs/getting-started.md. Using a dev dependency that requires PHP 7.2 seems reasonable.

Reply:
If we require php 7.2 to build it locally, then why not just install via composer then? When it there a time we are doing to build this plugin in 5.6?

Reply:
The problem is essentially that Composer does not distinguish between PHP version requirements for development and production dependencies. (i.e. it only supports `config.platform`, not `config.platform-dev`)

For development we can require 7.2+ (or whatever we find reasonable).

But for production we require PHP 5.6, as that's the PHP version supported by the plugin. That's why the platform is set to 5.6 here:

https://github.com/google/web-stories-wp/blob/8ba402752ebe6c01d9df3aee18d5b655a64c1b0e/composer.json#L24-L26

Let's say we want to add a feature to the WordPress plugin that is provided by "acme/super-dependency" (happens to require PHP 7.2+)  and we run `composer require acme/super-dependency`. Composer will thankfully tell us: "Hey, this requires PHP 7.2 but you still support 5.6".

That's what happens when you try to install phpstan via composer right now.

As mentioned elsewhere, one _could_ use `--ignore-platform-reqs` to suppress the above message and make it work, but that requires us to _always_ to use `--ignore-platform-reqs` when installing/updating/requiring packages to suppress the error message, so I'll end up doing `composer require acme/super-dependency --ignore-platform-reqs` too and now I'll crash everyone's PHP 5.6 site because of that.

Same happens when I just bump `config.platform`. I run `composer require acme/super-dependency`, it installs without errors, but I will not notice that it will break on PHP 5.6 sites.

Reply:
Fair enough. We can always tweak this after.